Libplanet/PoS/Control/Bond.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using Libplanet.Action;
+using Libplanet.Assets;
+
+namespace Libplanet.PoS
+{
+    internal static class Bond
+    {
+        internal static (IAccountStateDelta, FungibleAssetValue) Execute(
+            IAccountStateDelta states,
+            FungibleAssetValue consensusToken,
+            Address validatorAddress,
+            Address delegationAddress)
+        {
+            // TODO: Failure condition
+            // 1. Validator does not exist
+            // 2. Exchange rate is invalid(validator has no tokens but there are outstanding shares)
+            // 3. Amount is less than the minimum amount
+            // 4. Delegator does not have sufficient consensus token (fail or apply maximum)
+            if (!consensusToken.Currency.Equals(Asset.ConsensusToken))
+            {
+                throw new ArgumentException();
+            }
+
+            Validator validator = ValidatorCtrl.GetValidatorByAddr(states, validatorAddress);
+
+            // If validator share is zero, exchange rate is 1
+            // Else, exchange rate is validator share / token
+            FungibleAssetValue issuedShare
+                = ValidatorCtrl.ShareFromConsensusToken(states, validator.Address, consensusToken)
+                ?? throw new InvalidOperationException();
```

Comment:
`null` return is code that may cause unintended behavior. How about throwing an exception in the `ValidatorCtrl.ShareFromConsensusToken`?

Reply:
I've made it like it because I thought there can be some cases we need to know if `Validator` does not have `ConsensusToken`, and use this information.
Actually, I've thought exception is just bound to `bond` logic, not `ShareFromConsensusToken` converting logic.
If we throw error on `ShareFromConsensusToken`, we can handle our logic with try-catch on above condition, but I thought handling with null would be better like `getState`.

Reply:
Then how about this?
```suggestion
            if (!(ValidatorCtrl
                    .ShareFromConsensusToken(
                        states,
                        validator.Address,
                        consensusToken) is { } issuedShare))
            {
                throw new InvalidOperationException();
            }

```

Reply:
And IMO this is an anti-pattern using the same type as `ShareToken` and `ConsensusToken`. It may cause confuse 😵‍💫

Reply:
Seems nice, but could you explain the difference of those?
I've seen so many lines written like that, so I think it's better and robust way,
but don't know why.

Reply:
Hmm, do we have to inherit FAV? actually, It's a bit confusing to me also, to use different FAVs in just one type. (not just `Share` and `ConsensusToken`, but also `GovernanceToken`)...But not sure there are better way to handle.

Reply:
verbally explained.

Reply:
Pattern matching applied on commit 8045f68c000f18f2e9b8608ae7c2b33ccfc96cd0, but Wrapping FAV wasn't applied yet. 
I've extract this issue as separated : https://github.com/planetarium/libplanet/issues/2267